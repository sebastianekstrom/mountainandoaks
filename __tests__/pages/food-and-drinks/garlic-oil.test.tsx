import React from "react";
import { render, screen } from "@testing-library/react";
import GarlicOil from "pages/food-and-drinks/garlic-oil";
import { useRouter } from "next/router";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("GarlicOil page", () => {
  beforeEach(() => {
    (useRouter as jest.Mock).mockReturnValue({
      asPath: "/",
      route: "/food-and-drinks/garlic-oil",
    });
  });

  it("renders title correctly", () => {
    render(<GarlicOil />);
    const title = screen.queryAllByText(/Garlic oil/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<GarlicOil />);
    const ingress = screen.queryAllByText(
      /This garlic infused oil is perfect/i
    );
    expect(ingress).toBeDefined();
  });

  it("renders the ingredients", () => {
    render(<GarlicOil />);
    const flourLabel = screen.queryAllByText(/Cloves of garlic/i);
    expect(flourLabel).toBeDefined();
  });

  it("renders the method steps", () => {
    render(<GarlicOil />);
    const step = screen.queryAllByText(/Peel the garlic and crush the cloves/i);
    expect(step).toBeDefined();
  });
});
