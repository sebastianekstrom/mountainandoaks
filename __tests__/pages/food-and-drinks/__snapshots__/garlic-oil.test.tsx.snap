// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GarlicOil page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Cloves of garlic
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          10
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Rapeseed oil
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          160
          g
        </p>
      </li>
    </ul>
  </div>
</section>
`;
